app/views/tasks/index.html.haml

```diff
@@ -1,22 +1,22 @@
 %h1 Tasks
 
+= form_tag root_path, method: :get, remote: true do
+  = label_tag "Filter by:"
+  = select_tag "filter_tasks", options_for_select(Task::FILTER_TASKS_BY_STATES.keys)
+  = submit_tag 'Submit'
+
 %table
   %thead
     %tr
       %th Name
       %th Completed
+      %th List
       %th
       %th
       %th
 
-  %tbody
-    - @tasks.each do |task|
-      %tr
-        %td=task.name
-        %td=task.completed
-        %td=link_to 'Show', task
-        %td=link_to 'Edit', edit_task_path(task)
-        %td=link_to 'Destroy', task, :method => :delete, :data => { :confirm => 'Are you sure?' }
+  %tbody#tasks-table
+    = render 'table', task: @tasks
```

Comment:
i moved the table into it's own partial so it could be leveraged within `app/views/tasks/index.js.erb`